staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go

```diff
@@ -232,6 +245,14 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 // Unexported field members cannot be compared and will cause an informative panic; you must add an Equality
 // function for these types.
 func (e Equalities) DeepEqual(a1, a2 interface{}) bool {
+	return e.deepEqual(a1, a2, true)
+}
+
+func (e Equalities) DeepEqualWithoutImplicitEqualities(a1, a2 interface{}) bool {
```

Comment:
I actually would have made this a property of the `Equalities` object if possible (might be tricky because of how we define the object), that avoids having to pass-in the parameter throughout the callstack, it gets copied automatically when you copy the semantic (we don't want to in that case, but I think it's reasonable in general).

Reply:
Unfortunately your suspicion is correct that it is tricky because of how the object is defined.